cylc/flow/data_store_mgr.py

```diff
@@ -925,63 +894,76 @@ def update_task_proxies(self, updated_tasks=None):
                     PbTask(id=t_id)).MergeFrom(t_delta)
                 tasks[t_id].MergeFrom(t_delta)
 
-    def update_family_proxies(self, cycle_points=None):
-        """Update state of family proxies.
-
-        Args:
-            cycle_points (list):
-                Update family-node state from given list of
-                valid cycle point strings.
-
-        """
-        family_proxies = self.data[self.workflow_id][FAMILY_PROXIES]
-        if cycle_points is None:
-            cycle_points = self.delta_cycle_states.keys()
-        if not cycle_points:
+    def update_family_proxies(self):
+        self.updated_state_families.clear()
+        while self.state_update_families:
+            for fam_id in self.state_update_families:
+                self._family_ascent_point_update(fam_id)
+                break
```

Comment:
Probably a better way of doing it .. However:
- `self.state_update_families` is modified within the recursive function, so can't loop on it directly.
- I just want to start on any member of `self.state_update_families`, it doesn't matter which one, and I want to keep it in the set (i.e. not use `pop()`)..

I'm open, if you have any better ideas.. But you can loop on a set you modify, and I don't want to make a copy of it, because that could mean covering the same ground twice..